src/node_jsvmapi.h

```diff
@@ -273,18 +252,46 @@ NODE_EXTERN napi_status napi_define_class(napi_env e,
                                           const napi_property_descriptor* properties,
                                           napi_value* result);
 
+// Methods to work with external data objects
+NODE_EXTERN napi_status napi_wrap(napi_env e,
+                                  napi_value jsObject,
+                                  void* nativeObj,
+                                  napi_finalize finalize_cb,
+                                  napi_ref* result);
+NODE_EXTERN napi_status napi_unwrap(napi_env e, napi_value jsObject, void** result);
+
+NODE_EXTERN napi_status napi_create_external(napi_env e,
+                                             void* data,
+                                             napi_finalize finalize_cb,
+                                             napi_value* result);
+NODE_EXTERN napi_status napi_get_value_external(napi_env e, napi_value v, void** result);
+
 // Methods to control object lifespan
-NODE_EXTERN napi_status napi_create_persistent(napi_env e, napi_value v,
-                                               napi_persistent* result);
-NODE_EXTERN napi_status napi_release_persistent(napi_env e, napi_persistent p);
-NODE_EXTERN napi_status napi_get_persistent_value(napi_env e, napi_persistent p,
-                                                  napi_value* result);
-
-NODE_EXTERN napi_status napi_create_weakref(napi_env e, napi_value v,
-                                            napi_weakref* result);
-NODE_EXTERN napi_status napi_get_weakref_value(napi_env e, napi_weakref w,
-                                               napi_value* result);
-NODE_EXTERN napi_status napi_release_weakref(napi_env e, napi_weakref w);
+
+// Set initial_refcount to 0 for a weak reference, >0 for a strong reference.
+NODE_EXTERN napi_status napi_create_reference(napi_env e,
```

Comment:
Obviously nothing stops a weak-referenced value from being GC'd. If you try to increment the refcount after that, then the call to `napi_reference_addref` will return an error. But typically if you're using refcounting, you will start with a refcount of one, then increment/decrement until you get to 0.